supportingami/sami-website
# Feat/db export

#### What does this PR do?
Due to the way strapi randomly assigned hashes to images, there is no easy way to share upload file content using the existing import-export plugin. This is due to the fact that file references are stored in a `files` table the plugin does not have access to.

In order to workaround this new scripts have been added the import and export directly from sqlite .db files, so that the data state on one machine can be replicated to another. This has been done in such a way as to ignore admin and strapi core tables, so only user-generated content and files are tracked. It also contains additional logic to help ensure collaborators have their sqlite auto-index tables synced up (for tables that increment id numbers sequentially, some numbers will be missing if one developer uploads but then deletes a file - this ensures all developers track from the same numbers)

#### How should this be manually tested?
- Backup existing data if required (export to json files)
- Import data `yarn scripts strapi db:import`
- Run app, should see data populated from import
- Update data
- Create new export `yarn scripts strapi db:export`
- Should see git repo data updated

#### Screenshots (optional)
**Export CLI** - shows the database tables to import
![image](https://user-images.githubusercontent.com/10515065/188329638-1e8390f6-e1fb-48a0-949a-b9ec30433c4d.png)

**Exported Data** - now includes files and link_tables so that linked content can be imported correctly
![image](https://user-images.githubusercontent.com/10515065/188329967-08922c60-0f53-4532-9c74-6ff98aeb470b.png)
